### PR TITLE
CoilReductionGadget->ChannelGadget Conversion

### PR DESCRIPTION
**Pull request by apd47**
Continuing #1087 , this PR converts CoilReductionGadget to a ChannelGadget using the conventions established in #1090 and #1094

### CODE REVIEW

File: gadgets/mri_core/CoilReductionGadget.cpp [hunk 1]
```diff
@@ -1,122 +1,86 @@
...
+    auto h = (context.header);
+    coils_in_ = h.acquisitionSystemInformation->receiverChannels ? *h.acquisitionSystemInformation->receiverChannels : 128;
 
-    CoilReductionGadget::~CoilReductionGadget() {
+    std::string coil_mask_int = coil_mask;
+
+    if (coil_mask_int.compare(std::string("")) == 0) {
+        if (coils_out <= 0) {
+            GDEBUG("Invalid number of output coils %d\n", coils_out);
+            // TODO: How to throw Gadget failures?
```
Comment (hansenms) [inline gadgets/mri_core/CoilReductionGadget.cpp:15]: Why can you not just throw a runtime_error?

File: gadgets/mri_core/CoilReductionGadget.cpp [hunk 2]
```diff
@@ -1,122 +1,86 @@
...
         dims_out[1] = coils_out_;
 
-        GadgetContainerMessage< hoNDArray<std::complex<float> > >* m3 =
-            new GadgetContainerMessage< hoNDArray<std::complex<float> > >();
-
-        try{ m3->getObjectPtr()->create(dims_out);}
-        catch (std::runtime_error &err){
-            GEXCEPTION(err,"Unable to create storage for reduced dataset size\n");
-            return GADGET_FAIL;
+        hoNDArray<std::complex<float>> m3 = hoNDArray<std::complex<float>>();
```
Comment (hansenms) [inline gadgets/mri_core/CoilReductionGadget.cpp:60]: Remove `m1`, etc. naming.

File: gadgets/mri_core/CoilReductionGadget.cpp [hunk 3]
```diff
@@ -1,122 +1,86 @@
...
+        try {
+            m3.create(dims_out);
+        } catch (std::runtime_error& err) {
+            GEXCEPTION(err, "Unable to create storage for reduced dataset size\n");
```
Comment (hansenms) [inline gadgets/mri_core/CoilReductionGadget.cpp:64]: Here you are throwing, right?

**Review by hansenms — CHANGES_REQUESTED** [review]

File: gadgets/mri_core/CoilReductionGadget.cpp (hunk 1 above)
Comment (apd47) [reply gadgets/mri_core/CoilReductionGadget.cpp:15]: I just realized many of these TODO comments are from when I first started this conversion process a few months back and was less familiar with the Gadgetron log/error approaches - at least they're doing their job haha. Updating to using exception

**Review by hansenms — APPROVED** [review]

**Pull request merged**